# Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
# Проверьте его работу на данных, вводимых пользователем. При вводе пользователем нуля в качестве делителя
# программа должна корректно обработать эту ситуацию и не завершиться с ошибкой.

import traceback


class OwnError(Exception):
    def __str__(self):
        return 'На ноль делить нельзя же'


try:
    a = int(input('Введите делимое: '))
    b = int(input('Введите делитель: '))
    if b == 0:
        raise OwnError
    res = a / b
except ValueError:
    print('Числа вводить нужно')
except OwnError as err:
    print(err)
else:
    print(f'Результат деления: {res}')
